plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterPrice.php

```diff
@@ -120,144 +120,73 @@ protected function render( $attributes, $content, $block ) {
 			return '';
 		}
 
-		$price_range         = $this->get_filtered_price( $block );
-		$min_range           = $price_range['min_price'] ?? 0;
-		$max_range           = $price_range['max_price'] ?? 0;
-		$min_price           = intval( get_query_var( self::MIN_PRICE_QUERY_VAR, $min_range ) );
-		$max_price           = intval( get_query_var( self::MAX_PRICE_QUERY_VAR, $max_range ) );
-		$formatted_min_price = wc_price( $min_price, array( 'decimals' => 0 ) );
-		$formatted_max_price = wc_price( $max_price, array( 'decimals' => 0 ) );
-
-		$data = array(
-			'minPrice' => $min_price,
-			'maxPrice' => $max_price,
-			'minRange' => $min_range,
-			'maxRange' => $max_range,
+		$price_range = $this->get_filtered_price( $block );
+		$min_range   = $price_range['min_price'] ?? 0;
+		$max_range   = $price_range['max_price'] ?? 0;
+		$min_price   = intval( get_query_var( self::MIN_PRICE_QUERY_VAR ) ) ?: $min_range;
+		$max_price   = intval( get_query_var( self::MAX_PRICE_QUERY_VAR ) ) ?: $max_range;
```

Comment:
I believe we can limit the range that comes from the query params:

```suggestion
		$min_price   = intval( get_query_var( self::MIN_PRICE_QUERY_VAR ) ) ?: $min_range;
		$max_price   = intval( get_query_var( self::MAX_PRICE_QUERY_VAR ) ) ?: $max_range;
		
		$min_price = max( $min_range, min( $min_price, $max_range ) );
		$max_price = max( $min_range, min( $max_price, $max_range ) );
```

If this is an accepted change, I believe we would have to do the same within the `register_active_filters_data` method on lines 66 and 67.

Reply:
Yeah, we can do that, but we still need to redirect the page to the correct URL, or the client interactivity API will do it as it's doing now (update the URL and filter state after some seconds). I think it's better to work on it in another PR.